packages/ag-charts-community/src/chart/axis/axisTickGenerator.ts

```diff
@@ -333,26 +355,33 @@ export class AxisTickGenerator<S extends Scale<D, number, TickInterval<S>>, D> {
         primaryTickCount?: number;
     }): TickData {
         const { axis } = this;
-        const { nice, range, scale, visibleRange, interval } = axis;
+        const { range, scale, interval } = axis;
         const idGenerator = createIdsGenerator();
 
-        const tickParams: ScaleTickParams<any> = {
-            nice,
+        const domainParams: ScaleTickParams<any> = {
+            nice: niceMode === NiceMode.TickAndDomain,
             interval: interval.step,
             tickCount,
             minTickCount,
             maxTickCount,
         };
 
-        let niceDomain = nice ? scale.niceDomain(tickParams, domain) : domain;
+        const tickParams = {
+            ...domainParams,
+            nice: niceMode === NiceMode.TickAndDomain || niceMode === NiceMode.TicksOnly,
+        };
+
+        let niceDomain = niceMode === NiceMode.TickAndDomain ? scale.niceDomain(domainParams, domain) : domain;
 
+        let tickDomain: D[] = niceDomain;
         let rawTicks: any[];
 
         // @todo(xxx) - removing this references makes TS errors
         const scaleStopTsComplaining = scale;
```

Comment:
Nit: Is this still needed, I tried removing it locally and don't see any TS errors?